src/settings.cmake

```diff
@@ -216,14 +223,12 @@ endif()
 # containing the reference instead of using definitions from other modules.
 if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
     set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Xlinker -Bsymbolic -Bsymbolic-functions")
-    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--build-id=sha1")
-    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--build-id=sha1")
```

Comment:
The CMAKE_EXE_LINKER_FLAGS need to stay here and below, otherwise the host binaries would miss the build id sha1 in them. 
Also, the `-Wl,-z,relro,-z,now` should be added to this one too.